core_java/twitter/src/main/java/ca/jrvs/apps/twitter/impl/TwitterService.java

```diff
@@ -0,0 +1,145 @@
+package ca.jrvs.apps.twitter.impl;
+
+import ca.jrvs.apps.twitter.dao.CrdDao;
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.service.Service;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.util.StringUtils;
+
+@org.springframework.stereotype.Service
+public class TwitterService implements Service {
+
+  private CrdDao dao;
+
+  /**
+   * Constructor generates a TwitterService object
+   * that requires a CrdDao object to be passed in,
+   * most likely a TwitterDao object.
+   *
+   * @param dao
+   */
+  public TwitterService(CrdDao dao){this.dao = dao;}
+
+  /**
+   * Checks if text and coordinates do not violate
+   * Twitter constraints.
+   *
+   * @param tweet
+   * @return true if valid, false otherwise
+   */
+  private boolean validateTweet(Tweet tweet){
+
+    String text = tweet.getText();
+    float lon = tweet.getCoordinates().getCoordinates()[0];
+    float lat = tweet.getCoordinates().getCoordinates()[1];
+
+    return text.length() <= 140 && (lon <= 180 && lon >= -180) && (lat <= 90 && lat >= -90);
+
+  }
+
+  /**
+   * Checks if id is numerical sequence
+   *
+   * @param id
+   * @return true if numerical sequence, false otherwise
+   */
+  private boolean validateId(String id){
+
+    return id.matches("[0-9]+");
+
+  }
+
+  private void filterByField(Tweet tweet, String[] fields){
+
+    StringBuilder messageBuilder = new StringBuilder();
+
+    messageBuilder.append("{\n");
+
+    for (String field : fields){
+
+      messageBuilder.append("\"" + field + "\": ");
+
+      Method methodName = null;
+
+      try{
+        methodName = Tweet.class.getMethod("get" + StringUtils.capitalize(field));
+      }catch (NoSuchMethodException e){
+        throw new RuntimeException("The field " + field + " is an invalid field.");
+      }
+
+      Object result = null;
+       try{
+         result = methodName.invoke(tweet);
+       } catch (IllegalAccessException e) {
+         throw new RuntimeException("Permissions error.");
+       } catch (InvocationTargetException e2){
```

Comment:
`InvocationTargetException e`

`e2` looks weird to me.  just use `e` or `ex`

Reply:
the previous `e` is out of scope..so it's safe to use `e` again for the second execpetion